blockstore/apps/bundles/models.py

```diff
@@ -0,0 +1,186 @@
+impact overall, and we're never going to do range queryies on UUIDs anyhow. Side
+note: If you use UUID1 and get fancy with bit re-ordering, you can get around
+this, but it adds complexity to the code and prevents us from using the randomly
+generated UUID4.
+
+3) Another reason we're keeping UUIDs out of the primary key is because all
+secondary keys in MySQL's InnoDB storage engine store a copy of the primary key
+entry as well. A bigint is 8 bytes, and even an optimally packed UUID is still
+16 bytes.
+
+4) Our repos generally have VARCHAR(255) everywhere because MySQL's "utf8"
+encoding uses a max of three bytes per character (note: meaning it's not really
+UTF-8), and 255 * 3 = 765 bytes. The largest size for any column in an InnoDB
+index is 767 bytes. However, if we use MySQL's "utf8mb4", a.k.a. "actually
+UTF-8", that limit goes down to 191 bytes (191 * 4 = 764). Because 191 is a
+peculiar and difficult to remember number for an API, I am somewhat arbitrarily
+using 180 as a reasonable sounding number below the limit.
+
+5) In the case where there is a varchar field that needs to be unique within a
+certain context but might legitimately go over the 191 character limit, we can
+instead make a new column to store the md5 hash of the file path and then we can
+put a unique constraint on the tuple of (bundle_version, file_path_md5) giving
+more or less the same effect.
+
+6) There are a number of models that are made bigint when they don't strictly
+need to be (Bundles are unlikely to grow to that size, for instance).
+"""
+import uuid
+
+from django.db import models
+from django.dispatch import receiver
+
+from .data.store import BundleVersionKey, snapshot_created
+
+MAX_CHAR_FIELD_LENGTH = 180
+
+
+class Collection(models.Model):
+    id = models.BigAutoField(primary_key=True)
+    uuid = models.UUIDField(default=uuid.uuid4, unique=True)
+    title = models.CharField(max_length=MAX_CHAR_FIELD_LENGTH, db_index=True)
+
+    def __str__(self):
+        return "{} - {}".format(self.uuid, self.title)
+
+
+class Bundle(models.Model):
+    id = models.BigAutoField(primary_key=True)
+
+    # Collections are not just an organizational tool, but also determine
+    # ownership, licensing, permissions, etc. So we expect to only belong to one
+    # Collection at any given time.
+    #
+    # Comment for now just to make testing easier.
+#    collection = models.ForeignKey(
+#        Collection, related_name="bundles", related_query_name="bundle"
+#    )
+
+    # For pretty URLs?
+    slug = models.SlugField(allow_unicode=True)
+
+    title = models.CharField(max_length=MAX_CHAR_FIELD_LENGTH, db_index=True)
+    description = models.TextField(max_length=10_000)
+
+    # We use a BinaryField(16) because Django's built-in UUID field is double
+    # the size it needs to be (32 vs. 16 bytes) on MySQL. We use url-safe base64
+    # encoding when passing around serialized references.
+    uuid = models.UUIDField(default=uuid.uuid4, unique=True)
```

Comment:
I imagine you were already thinking of these but:

This says we use `BinaryField(16)` but it looks to be using the standard char storage?

Should we have a mixin that defines these `id BigAutoField, uuid UUIDField` fields and move the explanations about MySQL's limitations to that file?